h/api/auth.py

```diff
@@ -13,3 +15,21 @@ def get_user(request):
                 consumer = Consumer(key)
                 return User(userid, consumer, False)
     return None
+
+
+def translate_annotation_principals(principals):
+    """
+    Translate a list of annotation principals to a list of pyramid principals.
+    """
+    result = set([])
+    for principal in principals:
+        # Ignore suspicious principals from annotations
+        if principal.startswith('system.'):
+            continue
+        if principal == 'group:__world__':
+            result.add(security.Everyone)
```

Comment:
Hmm. This means that if I use the API to make an annotation with `'update': ['group:__world__']` then even people who aren't logged-in can edit the annotation. If we changed it to `security.Authenticated` here then they'd at least need to have an account. Or maybe this should be done at create time - don't allow `'group:__world__'` in the update permissions, or don't have the client sending the permissions at all!